odoo/cli/module.py

```diff
@@ -0,0 +1,152 @@
+            description="Upgrade selected modules",
+        )
+        self.uninstall_parser = subparsers.add_parser(
+            'uninstall',
+            help="Uninstall modules",
+            description="Uninstall selected modules",
+        )
+
+        for parser in (
+            self.install_parser,
+            self.uninstall_parser,
+            self.force_demo_parser,
+            self.upgrade_parser,
+        ):
+            parser.add_argument(
+                '-c', '--config', dest='config',
+                help="use a specific configuration file")
+            parser.add_argument(
+                '-d', '--database', dest='db_name', default=None,
+                help="database name, connection details will be taken from the config file")
+
+        self.install_parser.add_argument(
+            'modules', nargs='+', metavar='MODULE',
+            help="modules to be installed")
+        self.install_parser.epilog = textwrap.dedent("""\
+            Before installing modules, an Odoo database needs
+            to be created and initialized on your PostgreSQL
+            instance, using the `db create` command:
+
+            $ odoo-bin db create <db_name>
+
+            To get help on its parameters, see:
+
+            $ odoo-bin db create --help
+        """)
+        self.uninstall_parser.add_argument(
+            'modules', nargs='+', metavar='MODULE',
+            help="modules to be uninstalled")
+        self.upgrade_parser.add_argument(
+            'modules', nargs='+', metavar='MODULE',
+            help="modules to be upgraded")
+
+    def run(self, cmdargs):
+        parsed_args = self.parser.parse_args(args=cmdargs)
+
+        config_args = []
+        if parsed_args.config:
+            config_args += ['-c', parsed_args.config]
+        if parsed_args.db_name:
+            config_args += ['-d', parsed_args.db_name]
+        config.parse_config(config_args, setup_logging=True)
+
+        db_names = config['db_name']
+        if not db_names or len(db_names) > 1:
+            self.parser.error("Please provide a single database in the config file")
+        parsed_args.db_name = db_names[0]
+        if 'modules' in parsed_args:
+            parsed_args.modules = self._get_module_names(parsed_args.modules)
+
+        match parsed_args.subcommand:
+            case 'install':
+                self._install(parsed_args)
+            case 'uninstall':
+                self._uninstall(parsed_args)
+            case 'upgrade':
+                self._upgrade(parsed_args)
+            case 'force-demo':
+                self._force_demo(parsed_args)
+
+    def _get_module_names(self, module_names):
+        """ Get valid module names from disk before starting the Db environment """
+        initialize_sys_path()
+        module_names = set(module_names)
+        valid_module_names = {
+            module
+            for module in module_names
+            if get_module_path(module, display_warning=False)
```

Comment:
I guess it will have problem for `imported modules` mainly for `uninstall`

Reply:
I get what you mean. Any examples, context, alternatives?